google/ggrc-core
# GGRC-7802 Invalid Assessment states can be created via REST API

# Issue description

Invalid Assessment states can be created via REST API

# Steps to test the changes

Case 1:
1. Create assessment
2. Set verifiers
3. Run the put query with {'status': 'Completed'}

Case 2:
1. Create assessment
2. Run the put query with {"status": "In Review/Verify/Rework Needed"}

# Solution description

1. Added if statement to the common put method
2. Created `validate_done_state` method to validate assessment state

# Sanity checklist




# PR Review checklist
